tests/test_ess_csc.py

```diff
@@ -207,8 +241,126 @@ async def test_bin_script(self) -> None:
+            sensor_name=sensor_name,
+        )
+        reply = create_reply_dict(
+            sensor_name=sensor_name,
+            additional_data=[
+                topics_data["tel_relativeHumidity"].relativeHumidity,
+                topics_data["tel_temperature"].temperature[0],
+                topics_data["tel_dewPoint"].dewPoint,
+            ],
+        )
+        mtt = MockTestTools()
+        mtt.check_hx85a_reply(reply=reply, name=sensor_name)
+
+    async def validate_hx85ba_telemetry(
+        self, device_config: types.SimpleNamespace, sensor_name: str
+    ) -> None:
+        topics_data = await self.next_data(
+            topics=[
+                self.remote.tel_relativeHumidity,
+                self.remote.tel_temperature,
+                self.remote.tel_pressure,
+                self.remote.tel_dewPoint,
+            ],
+            sensor_name=sensor_name,
+        )
+        for attr_name in ("tel_temperature", "tel_pressure"):
+            assert topics_data[attr_name].numChannels == 1
+        reply = create_reply_dict(
+            sensor_name=sensor_name,
+            additional_data=[
+                topics_data["tel_relativeHumidity"].relativeHumidity,
+                topics_data["tel_temperature"].temperature[0],
+                topics_data["tel_pressure"].pressure[0] / csc.PASCALS_PER_MILLIBAR,
+                topics_data["tel_dewPoint"].dewPoint,
+            ],
+        )
+        mtt = MockTestTools()
+        mtt.check_hx85ba_reply(reply=reply, name=sensor_name)
+
+    async def validate_csat3b_telemetry(
+        self, device_config: types.SimpleNamespace, sensor_name: str
+    ) -> None:
+        data = await self.remote.tel_airTurbulence.next(
+            flush=False, timeout=STD_TIMEOUT
+        )
+        assert data.location == device_config.location
+        recordCounter = 1  # arbitrary value in range [0, 63]
+        status = 0
+        input_str = (
+            f"{data.speed[0]}{data.speed[1]}{data.speed[2]},"
+            f"{data.sonicTemperature},{status},{recordCounter}"
+        )
+        signature = common.sensor.compute_signature(input_str, ",")
+        reply = create_reply_dict(
+            sensor_name=data.sensorName,
+            additional_data=[
+                data.speed[0],
+                data.speed[1],
+                data.speed[2],
+                data.sonicTemperature,
+                status,
+                signature,
+            ],
+        )
+        mtt = MockTestTools()
+        mtt.check_csat3b_reply(reply=reply, name=sensor_name)
+
+    async def validate_windsonic_telemetry(
+        self, device_config: types.SimpleNamespace, sensor_name: str
+    ) -> None:
+        data = await self.remote.tel_airFlow.next(flush=False, timeout=STD_TIMEOUT)
+        assert data.location == device_config.location
+        # The check_windsonic_reply function in the ESS Common test utils only
+        # validates the speed and direction and not the other values.
```

Comment:
What other values? I can see how it could validate speed and direction, but looking at this code and the code for check_windsonic_reply, I don't see any attempt to send additional data that would be ignored, or anything like that. Maybe I missed something?

Reply:
I updated the doc string so it refers to standard deviation and mean.